### PR TITLE
Dark Skies Above

### PR DESCRIPTION
**Pull request by Zlorthishen**
#### Summary
Content "Adds an alien invasion force to game"

#### Purpose of change
This version changes the focus of the original mod from being a total conversion mod, to an additional content mod. It also updates the mod to keep code from becoming obsolete and allow for additional future content. 

#### Describe the solution
The mod adds an additional zombie faction along with an alien faction (The Xenoids) and some associated fauna. It also adds some weapons, armor and other supplies associated with the Xenoids, as well as their alien materials. 

#### Describe alternatives you've considered
I'd considered having the aliens friendly with the default zombie faction, however having them hostile against all life is appealing.

#### Testing
Game balance and other issues have been tested with many characters being sacrificed to please our Xenoid overlords.

### CODE REVIEW

**Comment by Mom-Bun** [comment]
Might want to remove the tileset files, that is part of undead people and has copyright issues

**Comment by Fris0uman** [comment]
> a total conversion mod, to an additional content mod.

this seems way less interesting than the initial intent

**Comment by Zlorthishen** [comment]
> Might want to remove the tileset files, that is part of undead people and has copyright issues

resolved.

File: data/mods/Dark-Skies-Above/DESIGN.md [hunk 1]
```diff
@@ -2,42 +2,41 @@
...
+The Earth presented in Dark Skies is a little closer to our current tech level (things you'd reasonably find in the 21st century). It is still roughly "some time in the near future", and some of the advancements in technology reflect this. The effects of global warming are already starting to increase the severity and frequency of natural disasters, COVID-21 rages unchecked while tensions between trade leaders are starting to grow due to global economic depression. As if things couldn't get any worse, astronomical institutions across the globe one day detected a peculiar gravitational anomaly just outside the Moon's orbit, initially considered a quaint discovery before alien spacecraft begin to spill from it...
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/DESIGN.md:23]: The design document mentions Covid-21 out of nowhere, and I dunno the thoughts on mentioning current day things like that which can get incredibly dated quickly.

File: data/mods/Dark-Skies-Above/DESIGN.md [hunk 2]
```diff
@@ -2,42 +2,41 @@
...
+In truth, the aliens themselves are a part of a rampant artificial intelligence network dedicated to acquiring resources from other planets by way of tunnelling through the multi-verse. The civilization that originally made it is long dead, and any interesting, vaguely religious aesthetics revering technology are essentially reflections of its birth culture. The network concerns itself with little more than expanding its scope, extracting resources, and refining its tools - though it occasionally will show small glimmers of sapience. It uses a mix of drones and cyborgs made from formerly conquered species to do its bidding, with the ultimate intent of turning planets it comes across to balls of rock hosting innumerable farms and factories. Think a mix between the AI from "Universal Paperclips", the Combine from Half-Life 2, and a little bit of The Tyranid Hivemind.
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/DESIGN.md:32]: Why the change from Cogmind to Tyranids? What is the connection of an AI super-network to a gooey hivemind that eats everything?

File: data/mods/Dark-Skies-Above/DESIGN.md [hunk 3]
```diff
@@ -2,42 +2,41 @@
...
+Edit: You can also contact me, '@Zlorthishen'.  I have a few ideas for directions for this mod which aren't listed in this file. Expect a free season pass DLC named "Peach Trees" soon™.
 
 
-# Contributing
-
-Since Dark Skies is distributed with the base game, contributions are welcome. If you have any questions, you can contact me on Discord via `@ephemeral_storyteller` through the official developer or official community severs.
+# Summary of Intent and Content
 
+Mechanically, Dark Skies Above began as a project to demonstrate how you can push the game engine to present different stories. Unlike the base game, a narrative arc is assumed: the player will want find a way to make occupation of Earth too costly for the aliens to continue, or die trying. This is, of course, by no means an "easy" or "short" process, nor is it necessarily a requirement. This is done largely to compensate for the fact that Dark Skies does not play well with other mods or some aspects of active development, thus needing a tighter experience.
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/DESIGN.md:14]: Considering this is explicitly moving from conversion to just a faction/content mod, this should be removed.

File: data/mods/Dark-Skies-Above/DESIGN.md [hunk 4]
```diff
@@ -2,42 +2,41 @@
 
 This document is intended to not only guide new potential contributors, but also help get my own thoughts on how the game will progress in text. Assume spoilers beyond this point!
 
+# Contributing
 
-# Summary of Intent and Content
-
-Mechanically, Dark Skies Above began as a project to demonstrate how you can push the game engine to present different stories. Unlike the base game, a narrative arc is assumed: the player will want find a way to make occupation of Earth too costly for the aliens to continue, or die trying. This is, of course, by no means an "easy" or "short" process, nor is it necessarily a requirement. This is done largely to compensate for the fact that Dark Skies does not play well with other mods or some aspects of active development, thus needing a tighter experience.
+Since Dark Skies is distributed with the base game, contributions are welcome. If you have any questions, you can contact me on Discord viazz `@ephemeral_storyteller` through the official developer or official community severs.
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/DESIGN.md:7]: Ephemeral Storyteller is no longer the maintainer, and should probably be removed from "contacting"

File: data/mods/Dark-Skies-Above/README.md [hunk 5]
```diff
@@ -1,16 +1,18 @@
 # Dark Skies Above
-A partial conversion mod for Cataclysm: DDA, taking inspiration from XCOM 2, Half Life 2, and similar 'alien invasion' media to present a game of survival against an alien occupation force. WIP, but will feature new quests, NPCs, locations, and enemies.
+A partial conversion mod for Cataclysm: DDA, taking inspiration from XCOM 2, Half Life 2, the movie Skyline (not the sequel) and similar 'alien invasion' media to present a brutal game of survival against an alien expeditionary force. 
 
 # Takeaway Features
-In short, Dark Skies is more difficult than vanilla DDA, and reccomended for players who have some experience under their belt. 
+In short, this Dark Skies is much much more difficult than vanilla DDA, and recommended for players who have some experience under their belt.
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/README.md:5]: Just one "much" can do

File: data/mods/Dark-Skies-Above/README.md [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
+ - If you survive long enough, you may even see the Xenoid's emissaries to humanity.
+ 
+ -WIP Map: "Peach Trees" Arcology by Zenture Industries®
+ -WIP Map: The Alien Hive Structure
+ -WIP Map: The Xenoid Fortress
+ 
 
 # Meta
-Being a conversion mod, this mod won't work with most other mods. Likewise, tileset support is probably going to be patchy, however I'll be making good use of looks_like for your viewing pleasure.
+This is NOT a total conversion mod, I'll strive to make Dark Skies Above work as friendly with other mods as possible, and will try to maintain tileset support based off UndeadPeople tileset. I intend for this mod to challenge you as much as possible and won't pull any punches, this mod is people who want a crazy challenge and/or radical change to the base game. Having said that I really don't have any idea what I'm doing so keep that in mind before complaining. The original design document is included with the mod, put the specifics may not be included and should be considered apocryphal as far as future features.
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/README.md:18]: You should probably remove the original design document if its not longer applicable to the current vision of the mod.

File: data/mods/Dark-Skies-Above/harvest.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [
   {
-    "id": "dks_alien_cyborg",
+    "id": "dkz_alien_cyborg",
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/harvest.json:3]: Why is every ID changed from "dks" to "dkz"?

This doesn't make any sense, and not only that, it breaks saves due to no migration.

File: data/mods/Dark-Skies-Above/items/alien-scrap.json [hunk 8]
```diff
@@ -1,110 +1,124 @@
...
+    "container": "flask_glass",
     "comestible_type": "DRINK",
     "symbol": "~",
-    "quench": 5,
-    "description": "A lump of alien hydrogel with small writhing specks in it.  Useful in crafting.  You could 'drink' it, but there's no way it would be pleasant.",
+    "quench": -3,
+    "stim": 13,
+    "addiction_potential": 77,
+    "addiction_type": "opiate",
+    "description": "A small blue alien liquidy-gel with small writhing grey specks in it.  You could eat it if you want.  Given how you've acquired it, you aren't sure that it would be a good idea so you should probably save your game first.",
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/items/alien-scrap.json:104]: Not a fan of the fourth wall breaking here.

File: data/mods/Dark-Skies-Above/items/electronics.json [hunk 9]
```diff
@@ -1,58 +1,216 @@
...
-    "name": { "str": "broken seeker drone" },
+    "name": { "str": "broken xenoid seeker drone" },
+    "material": [ "dkz_biogel_mat", "dkz_blend_mat", "dkz_biopoly_mat" ],
+    "flags": [ "TRADER_AVOID", "NO_REPAIR", "LEAK_DAM", "RADIOACTIVE" ],
     "description": "A broken drone.  Much less threatening now that it isn't prodding you.  Specialized alien tools would be best for disassembly, but you could make do with more human instruments instead."
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "id": "dkz_zenture_industries_basic_zwatch",
+    "name": { "str": "Zenture Industries® z-Ŵatch™", "str_pl": "Zenture Industries® z-Ŵatches™" },
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/items/electronics.json:79]: What is Zenture Industries? Why do they have 3 watches? Why is it so advance and "cool" that it the description goes out of its way to state how cool it is, how user friendly it is, and how advanced it is? And who is Mr. Zenture and team zenture?

File: data/mods/Dark-Skies-Above/items/weapons.json [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    "longest_side": "129 cm",
+    "price": 310000,
+    "price_postapoc": 7500,
+    "bashing": 17,
+    "cutting": 40,
+    "material": [ "dkz_blend_mat" ],
+    "symbol": "/",
+    "color": "light_gray",
+    "charges_per_use": 1,
+    "max_charges": 50,
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/items/weapons.json:20]: What charges does this use? It seems to not have a storage slot nor connect to UPS as far as I can tell.

File: data/mods/Dark-Skies-Above/modinfo.json [hunk 11]
```diff
@@ -2,11 +2,11 @@
   {
     "type": "MOD_INFO",
     "id": "darkskies",
-    "name": "Dark Skies Above",
-    "authors": [ "ephemeral_storyteller" ],
-    "description": "A total conversion that shifts the Cataclysm towards an alien occupation survival scenario.  THIS MOD WILL BREAK INTENDED FUNCTIONALITY OF OTHER MODS!  USE OTHER MODS AT YOUR OWN RISK.",
+    "name": "<color_light_green>Dark Skies Above</color>",
+    "maintainers": [ "Zlorthishen" ],
+    "authors": [ "Zlorthishen", "ephemeral_storyteller", "Kenan" ],
```
Comment (Mom-Bun) [inline data/mods/Dark-Skies-Above/modinfo.json:7]: Why is Kenan listed as an author here?

**Review by Mom-Bun — COMMENTED** [review]
Looking in-depth in this and it is... underwhelming to say the least, and incredibly disappointing. Dark Skies Above, as a mod, planned to do something interesting and very cool, to prove that Cataclysm, as a game wasn't just something for apocalyptic survival, but its engine itself could be used for other interesting projects, to show how far you can push it. 

This does not do any of these things, it just adds another faction, and poorly... written at that?

**Comment by Mom-Bun** [comment]
Also on top of these criticism, you are pulling from your master branch, which is bad practice in general, and seems you are making these edits via web instead of desktop side; I really suggest you check out the guides on setting up git on your desktop so you don't inflate this with too many commits.

**Comment by sayke** [comment]
This is fantastic, Zlorthishen - thank you so much for integrating it. Most players will never mess with a total conversion that strips out all the base elements of the game, but adding Xenoids as another element to the Cataclysm actually improves it. I will definitely use this on my 0.F playthrough.

**Comment by Mom-Bun** [comment]
> This is fantastic, Zlorthishen - thank you so much for integrating it. Most players will never mess with a total conversion that strips out all the base elements of the game, but adding Xenoids as another element to the Cataclysm actually improves it. I will definitely use this on my 0.F playthrough.

I disagree, many, many players are interested and would be willing to mess with a total conversion, its just total conversion mods are a lot of effort to maintain and so there is hard competition with the very expansive base game, but this doesn't mean players inherently hate it.

**Comment by kevingranade** [comment]
Based on this, "This version changes the focus of the original mod from being a total conversion mod, to an additional content mod." and the lack of backwards compatibility with the original DSA, I need you to shift to a new directory and change the name and mod I'd to avoid a collision. 
Basically this isn't dark skies above anymore, which is not a problem, but it needs to not say it's dark skies above.

File: data/mods/Dark-Skies-Above/DESIGN.md (hunk 1 above)
Comment (Zlorthishen) [reply data/mods/Dark-Skies-Above/DESIGN.md:23]: <adds back in snippets for Presidential Election Fliers>

File: data/mods/Dark-Skies-Above/DESIGN.md (hunk 2 above)
Comment (Zlorthishen) [reply data/mods/Dark-Skies-Above/DESIGN.md:32]: because i play Tyrannids and I like their inscrutable natural evilness

File: data/mods/Dark-Skies-Above/modinfo.json (hunk 11 above)
Comment (Zlorthishen) [reply data/mods/Dark-Skies-Above/modinfo.json:7]: because he helped

File: data/mods/Dark-Skies-Above/items/weapons.json (hunk 10 above)
Comment (Zlorthishen) [reply data/mods/Dark-Skies-Above/items/weapons.json:20]: removed

File: data/mods/Dark-Skies-Above/modinfo.json (hunk 11 above)
Comment (Mom-Bun) [reply data/mods/Dark-Skies-Above/modinfo.json]: Unmarking this as unresolved as this brings up another problem, if Kenan helped here, I see no mention of his contributions in the commit history, which is a problem.

**Comment by Mom-Bun** [comment]
I'm going to echo Kevin here, this needs to become its own mod with its own new name and directory considering it shares zero similarities with the mission statement of the original mod and is no longer dark skies above at all, I'd rather the old code sit unchanged in case a future maintainer wants to stick with the original mission or the original maintainer comes back as the idea behind Dark Skies Above was cool, and personally, I'd rather it not be cannibalized for another generic content mod.

**Comment by Termineitor244** [comment]
Dark Skies Above Reborn: #48132
